src/plugins/discover/common/embeddable/search_inject_extract.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { SavedObjectReference } from '@kbn/core-saved-objects-server';
+import type { EmbeddableStateWithType } from '@kbn/embeddable-plugin/common';
+import type { SearchByValueInput } from '@kbn/saved-search-plugin/public';
+
+export const inject = (
+  state: EmbeddableStateWithType,
+  injectedReferences: SavedObjectReference[]
+): EmbeddableStateWithType => {
+  if (hasAttributes(state)) {
+    // Filter out references that are not in the state
+    // https://github.com/elastic/kibana/pull/119079
```

Comment:
@ThomThomson It seems like we have to use the same approach to filtering references as the Lens `inject` function due to the same issue outlined in #119079. When we have a by value saved search using a temporary data view, the saved search has no references which seems to result in the Dashboard `inject` code passing in all dashboard references instead: https://github.com/elastic/kibana/blob/0d8a2da9792caef4171c31c71b3836ec5e44bdb4/src/plugins/dashboard/common/dashboard_container/persistable_state/dashboard_container_references.ts#L41-L45

Let me know if this makes sense to you or if I'm missing something here.

Reply:
Ah of course, that makes a lot of sense!